compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -52,6 +52,7 @@ class CompilationTests {
       compileFile("tests/pos-special/notNull.scala", defaultOptions.and("-Yexplicit-nulls")),
       compileDir("tests/pos-special/adhoc-extension", defaultOptions.and("-source", "3.1", "-feature", "-Xfatal-warnings")),
       compileFile("tests/pos-special/i7575.scala", defaultOptions.andLanguageFeature("dynamics")),
+      compileFile("tests/pos-special/i10511-transitivity.scala", defaultOptions), // avoid pickling test
```

Comment:
I think it would be better to add the new test to `compiler/test/dotc/pos-test-pickling.blacklist` were we already have several match type tests.

Reply:
the pickling test seemed to work after cleaning my repo so ive merged it with the other pos test for now